test/compress/annotations.js

```diff
@@ -0,0 +1,465 @@
+        inline: true,
+        keep_fargs: false,
+        reduce_vars: true,
+        sequences: true,
+        side_effects: true,
+        toplevel: true,
+        unused: true,
+    }
+    input: {
+        var f = function(a) {
+            return /*@__PURE__*/ function(b) {
+                console.log(b);
+            }(a);
+        };
+        var a = f("PASS");
+    }
+    expect: {}
+}
+
+inline_pure_call_3: {
+    options = {
+        annotations: true,
+        collapse_vars: true,
+        evaluate: true,
+        inline: true,
+        keep_fargs: false,
+        passes: 2,
+        reduce_vars: true,
+        toplevel: true,
+        unused: true,
+    }
+    input: {
+        var f = function(a) {
+            return /*@__PURE__*/ function(b) {
+                console.log(b);
+            }(a);
+        };
+        var a = f("PASS");
+        console.log(a);
+    }
+    expect: {
+        var a = function() {
+            console.log("PASS");
+        }();
+        console.log(a);
+    }
+    expect_stdout: [
+        "PASS",
+        "undefined",
+    ]
+}
+
+inline_pure_call_4: {
+    options = {
+        annotations: true,
+        evaluate: true,
+        reduce_vars: true,
+        toplevel: true,
+        unused: true,
+    }
+    input: {
+        var a = /*@__PURE__*/ function() {
+            return console.log("PASS"), 42;
+        }();
+        console.log(a);
+    }
+    expect: {
+        var a = function() {
+            return console.log("PASS"), 42;
+        }();
+        console.log(a);
+    }
+    expect_stdout: [
+        "PASS",
+        "42",
+    ]
+}
+
+compress_and_output_annotations_enabled: {
+    options = {
+        annotations: true,
+        side_effects: true,
+    }
+    beautify = {
+        annotations: true,
+        beautify: true,
+        comments: false,
+    }
+    input: {
+        var top = /*@__PURE__*/ foo();
+        /*@__PURE__*/ a(1)(2)(3);
+        /*@__PURE__*/ (b(1))(2)(3);
+        /*@__PURE__*/ (c(1)(2))(3);
+        /*@__PURE__*/ (d(1)(2)(3));
+        (/*@__PURE__*/ e)(1)(2)(3);
+        (/*@__PURE__*/ f(1))(2)(3);
+        (/*@__PURE__*/ g(1)(2))(3);
+        (/*@__PURE__*/ h(1)(2)(3));
+    }
+    expect_exact: [
+        "var top = /*@__PURE__*/foo();",
+        "",
+        "e(1)(2)(3);",
+        "",
+        "f(1)(2)(3);",
+        "",
+        "g(1)(2)(3);",
+    ]
+}
+
+compress_annotations_disabled_output_annotations_enabled: {
+    options = {
+        annotations: false,
+        evaluate: true,
+        sequences: true,
+        side_effects: true,
+    }
+    beautify = {
+        annotations: true,
+        comments: true,
+    }
+    input: {
+        /*@__PURE__*/ a(1+2);
+        /*#__PURE__*/ (b(2+3));
+        (/*@__PURE__*/ c)(side_effect);
+        (/*#__PURE__*/ d(effect()));
+    }
+    expect_exact: [
+        "/*@__PURE__*/a(3),",
+        "/*#__PURE__*/b(5),",
+        "c(side_effect),",
+        "/*#__PURE__*/d(effect());",
```

Comment:
With regards to https://github.com/mishoo/UglifyJS/pull/4758#issuecomment-794674867, I've decided to just copy the annotation (style) from the input instead of forcing it one way or another.